### DISCUSSION TITLE
Testing Comment Route API

### DISCUSSION THREAD

**Discussion by manojtsx** [manojtsx/GyanShristi-CMS#49] [Announcements]
1. `POST /api/comment/add` - Comment on the content.
- On successful comment creation, it shows "Comment done" (200).
- If the user is not found, it shows "You may not be logged in for comment" (404).
- If the parent comment is not found, it shows "Parent comment not found" (401).
- If the parent comment is already a reply, it shows "Reply to the main comment only allowed" (400).
- If the user's role is not authorized, it shows "You must log in to comment" (401).
- If any other error occurs, it shows a server error (500).

3.  `PUT /api/comment/:id` - Edit the content.
- On successful comment update, it shows "Comment updated" (200).
- If the comment does not exist, it shows "Comment not found" (404).
- If the user is not authorized to edit the comment, it shows "You are not authorized to edit this comment" (403).
- If any other error occurs, it shows a server error (500).

5.  `GET /api/comment/comments` - Get comment by content
- On successful retrieval of comments, it shows "Comments retrieved" with the list of comments (200).
- If no comments are found for the given content, it shows "No comments found matching the content" (404).
- If any other error occurs, it shows a server error (500).

7.  `GET /api/comment/`  - Get all comments list
- On successful retrieval of comments, it shows "Comment retrieved" with the list of comments (200).
- If no comments are found, it shows "No comment found" (404).
- If the user is not authorized (not admin or editor), it shows "You are not authorized for this action." (401).
- If any other error occurs, it shows a server error (500).

9.  `DELETE /api/comment/delete/:id` - Delete comment
- On successful deletion, it shows "Comment deleted successfully" (200).
- If the user is not found, it shows "You may not be logged in for comment" (404).
- If the comment does not exist, it shows "Comment not found" (404).
- If the user is not authorized to delete the comment, it shows "You are not authorized to delete this comment" (403).
- If any other error occurs, it shows a server error (500).

Note : After every API call, don't forget to check the db to track changes.

**Comment by seezan25** [comment]
Tried to delete the comment by editor role through : http://localhost:3001/api/comment/delete/66b5e309db5148746b5e320a 

Note: The content id is correctly written

Problem: shows "You are not authorized to delete this comment" while editor should be authorized to delete comments

**Comment by seezan25** [comment]
Tried to delete own comment by the author through: http://localhost:3001/api/comment/delete/66b5e309db5148746b5e320a

Note: Both the user id and comment id are correct

Problem:  Shows "Comment not found" error
